source/modulo_controllers/src/ControllerInterface.cpp

```diff
@@ -489,16 +489,52 @@ void ControllerInterface::publish_predicates() const {
   predicate_publisher_->publish(message);
 }
 
+std::string ControllerInterface::validate_and_declare_signal(
+    const std::string& signal_name, const std::string& type, const std::string& default_topic, bool fixed_topic) {
+  auto parsed_signal_name = utilities::parse_topic_name(signal_name);
+  if (parsed_signal_name.empty()) {
+    RCLCPP_WARN(
+        get_node()->get_logger(),
+        "The parsed signal name for %s '%s' is empty. Provide a string with valid characters for the signal name "
+        "([a-zA-Z0-9_]).",
```

Comment:
nitpick: we may be more restrictive in a future schema to disallow uppercase signal names. If `parse_topic_name()` changes as a result of that or for any other reason, the warning text with valid characters will need to be updated correspondingly. I also assume that the component interface has an equivalent warning. Therefore it might be better to define a const string or constexpr function with the default validation warning in the utilities header and import it for use here

Reply:
You are right, this warning is repeated in modulo components. The effort to reduce code duplication is part of a different branch and stream of PRs, I'd prefer to this there. I've created an issue for this comment